### PR TITLE
Add error notification when Galaxy App transitions to Error [IA-2655]

### PR DESCRIPTION
**Pull request by ehanna4**
Working on getting Galaxy apps to more parity with Notebooks
- previuosly a transition to "Error" status would not trigger a  popup, now it will
Screen Shot 2021-04-08 at 4 39 13 PM

Possibly some code duplication in the error handling though

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]
🎉

File: src/components/RuntimeManager.js [hunk 1]
```diff
@@ -108,6 +110,50 @@ const RuntimeErrorNotification = ({ runtime }) => {
...
+export const AppErrorModal = ({ app, onDismiss }) => {
+  const [error, setError] = useState()
+  const [loadingAppDetails, setLoadingAppDetails] = useState(false)
+
+  const loadAppError = _.flow(
+    withErrorReporting('Error loading app details'),
+    Utils.withBusyState(setLoadingAppDetails)
+  )(async () => {
+    const { errors: appErrors } = await Ajax().Apps.app(app.googleProject, app.appName).details()
+    setError(appErrors[0].errorMessage)
```
Comment (jdcanas) [inline src/components/RuntimeManager.js]: should this be `appErrors[0]?.errorMessage` to guard against when there are none (one of those 'shouldn't happen but why not be safe' things)

File: src/components/RuntimeManager.js [hunk 2]
```diff
@@ -183,6 +229,11 @@ export default class RuntimeManager extends PureComponent {
           })
         ])
       })
+    } else if (prevApp && prevApp.status !== 'ERROR' && app && app.status === 'ERROR' && !_.includes(app.appName, errorNotifiedApps.get())) {
```
Comment (jdcanas) [inline src/components/RuntimeManager.js]: I think this should be `else if (app?.status === 'ERROR' && !_.includes(app.appName, errorNotifiedApps.get()))`. I think it's important to show the error on page load (aka if they navigate away when the error occurs and come back, or just open the page to an errored app).

**Review by jdcanas — APPROVED** [review]
Good change. Very annoying not to be able to see the error details. Few small changes I think are worth talking about.

File: src/components/RuntimeManager.js (hunk 2 above)
Comment (ehanna4) [reply src/components/RuntimeManager.js]: ah yeah i like that idea, with the runtime it saves it to the little error icon so they can reopen it but galaxy doesnt have that

**Pull request merged**